crates/bdk/src/descriptor/mod.rs

```diff
@@ -494,7 +512,7 @@ impl DescriptorMeta for ExtendedDescriptor {
             false
         });
 
-        path_found.map(|path| self.at_derivation_index(path))
+        path_found.map(|path| self.at_derivation_index(path).expect("TODO"))
```

Comment:
In d12242ec746ffc51da64672d7c7f7e014258d190:

This TODO can just be `unwrap` with a comment explaining that `path_found` came out of a child index and therefore is in range.